CI test for macos sometimes fails in test_it_uses_pip_repositories_with_env_var_substitution

### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

This only happens some of the time. Maybe some sort of race condition?

```
_ ERROR at teardown of test_it_uses_pip_repositories_with_env_var_substitution[conda] _
[gw3] darwin -- Python 3.12.1 /Users/runner/micromamba/envs/conda-lock-dev/bin/python3.12

    @pytest.fixture(scope="module")
    def private_package_tar():
        tar_path = _PRIVATE_PACKAGE_SDIST_PATH.parent / "fake-private-package-1.0.0.tar.gz"
        with tarfile.open(tar_path, "w:gz") as tar:
            tar.add(
                _PRIVATE_PACKAGE_SDIST_PATH,
                arcname=os.path.basename(_PRIVATE_PACKAGE_SDIST_PATH),
            )
        try:
            yield tar_path
        finally:
>           os.remove(tar_path)
E           FileNotFoundError: [Errno 2] No such file or directory: '/Users/runner/work/_temp/tests/test-pip-repositories/fake-private-package-1.0.0.tar.gz'

tests/test_pip_repositories.py:54: FileNotFoundError
```

### Conda Info

_No response_

### Conda Config

_No response_

### Conda list

_No response_

### Additional Context

_No response_

--- a/tests/test_pip_repositories.py
+++ b/tests/test_pip_repositories.py
@@ -35,23 +35,17 @@ _PRIVATE_REPO_PACKAGE = """<!DOCTYPE html>
 </html>
 """
 
-_PRIVATE_PACKAGE_SDIST_PATH = (
-    Path(__file__).parent / "test-pip-repositories" / "fake-private-package-1.0.0"
-)
 
-
-@pytest.fixture(scope="module")
-def private_package_tar():
-    tar_path = _PRIVATE_PACKAGE_SDIST_PATH.parent / "fake-private-package-1.0.0.tar.gz"
+@pytest.fixture
+def private_package_tar(tmp_path: Path):
+    sdist_path = (
+        clone_test_dir("test-pip-repositories", tmp_path) / "fake-private-package-1.0.0"
+    )
+    assert sdist_path.exists()
+    tar_path = sdist_path / "fake-private-package-1.0.0.tar.gz"
     with tarfile.open(tar_path, "w:gz") as tar:
-        tar.add(
-            _PRIVATE_PACKAGE_SDIST_PATH,
-            arcname=os.path.basename(_PRIVATE_PACKAGE_SDIST_PATH),
-        )
-    try:
-        yield tar_path
-    finally:
-        os.remove(tar_path)
+        tar.add(sdist_path, arcname=os.path.basename(sdist_path))
+    return tar_path
 
 
 @pytest.fixture(

